src/test/kotlin/org/taymyr/lagom/elasticsearch/search/ElasticSearchDateRangeIT.kt

```diff
@@ -0,0 +1,163 @@
+import org.taymyr.lagom.elasticsearch.CreationDateRange
+import org.taymyr.lagom.elasticsearch.LagomClientAndEmbeddedElastic
+import org.taymyr.lagom.elasticsearch.SampleDocument
+import org.taymyr.lagom.elasticsearch.SampleDocumentResult
+import org.taymyr.lagom.elasticsearch.deser.invoke
+import org.taymyr.lagom.elasticsearch.deser.invokeT
+import org.taymyr.lagom.elasticsearch.indices.dsl.CreateIndex
+import org.taymyr.lagom.elasticsearch.indices.dsl.MappingProperty
+import org.taymyr.lagom.elasticsearch.search.dsl.SearchRequest
+import org.taymyr.lagom.elasticsearch.search.dsl.query.compound.BoolQuery
+import org.taymyr.lagom.elasticsearch.search.dsl.query.term.DateRange
+import org.taymyr.lagom.elasticsearch.search.dsl.query.term.DateRangeExpression
+import org.taymyr.lagom.elasticsearch.search.dsl.query.term.RangeQuery
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+import java.util.UUID
+
+class ElasticSearchDateRangeIT : WordSpec() {
+
+    override fun listeners(): List<TestListener> = listOf(Companion)
+
+    init {
+        "ElasticSearch" should {
+            val indexName = "date-range-test"
+            val typeName = "samples"
+            "successfully create the index" {
+                val request = CreateIndex(
+                    CreateIndex.Settings(1, 1),
+                    mapOf(
+                        typeName to CreateIndex.Mapping(mapOf(
+                            "id" to MappingProperty.TEXT,
+                            "user" to MappingProperty.KEYWORD,
+                            "message" to MappingProperty.KEYWORD,
+                            "age" to MappingProperty.INTEGER,
+                            "balance" to MappingProperty.DOUBLE,
+                            "creationDate" to MappingProperty.DATE
+                        ))
+                    )
+                )
+                whenReady(elasticIndices.create(indexName).invoke(request).toCompletableFuture()) {
+                    it.acknowledged shouldBe true
+                    it.shardsAcknowledged shouldBe true
+                    it.index shouldBe indexName
+                }
+            }
+            "successfully index 24 documents with creationDate of 2018-01-01 UTC (one per hour)" {
+                repeat(24) {
+                    val creationDate = ZonedDateTime.now().withZoneSameInstant(ZoneOffset.UTC).truncatedTo(ChronoUnit.DAYS)
+                        .withYear(2018).withMonth(1).withDayOfMonth(1).withHour(it)
+                    val doc = SampleDocument(
+                        user = "thisDayUser-${UUID.randomUUID()}",
+                        message = "messageOfThisDay",
+                        creationDate = creationDate.toInstant().toEpochMilli()
+                    )
+                    whenReady(elasticDocument.indexWithId(indexName, typeName, UUID.randomUUID().toString())
```

Comment:
One `UUID.randomUUID()` for one document.

Reply:
it seems it's already one UUID per doc since the ivocation inside the `repeat`

Reply:
```kotlin
user = "thisDayUser-${UUID.randomUUID()}"
...
whenReady(elasticDocument.indexWithId(indexName, typeName, UUID.randomUUID().toString())
```

Reply:
ok, done